napari/layers/shapes/_shapes_utils.py

```diff
@@ -797,68 +788,79 @@ def triangulate_face(
+        tri = Triangulation(raw_vertices, edges)
+        tri.triangulate()
+        vertices, triangles = tri.pts, tri.tris
     except Exception as e:  # pragma: no cover
-        path = tempfile.mktemp(prefix='napari_triang_', suffix='.npz')
-        text_path = tempfile.mktemp(prefix='napari_triang_', suffix='.txt')
-        np.savez(path, data=polygon_vertices)
-        np.savetxt(text_path, polygon_vertices)
+        path, text_path = _save_failed_triangulation(
+            raw_vertices, backend='vispy'
+        )
         raise RuntimeError(
             f'Triangulation failed. Data saved to {path} and {text_path}'
         ) from e
 
+    triangles = triangles.astype(np.uint32)
+
+    return vertices, triangles
+
 
 @overload
-def _triangulate_face(
+def triangulate_face_triangle(
     raw_vertices: CoordinateArray2D,
     edges: EdgeArray,
     polygon_vertices: CoordinateArray,
 ) -> tuple[CoordinateArray2D, TriangleArray]: ...
 
 
 @overload
-def _triangulate_face(
+def triangulate_face_triangle(
     raw_vertices: CoordinateArray3D,
     edges: EdgeArray,
     polygon_vertices: CoordinateArray,
 ) -> tuple[CoordinateArray3D, TriangleArray]: ...
 
 
-def _triangulate_face(
+def triangulate_face_triangle(
     raw_vertices: CoordinateArray,
     edges: EdgeArray,
     polygon_vertices: CoordinateArray,
 ) -> tuple[CoordinateArray, TriangleArray]:
-    if triangulate is not None:
-        # if the triangle library is installed, use it because it's faster.
-        res = triangulate(
-            {'vertices': raw_vertices, 'segments': edges}, opts='p'
-        )
-        vertices = res['vertices']
-        raw_triangles = res['triangles']
-        # unlike VisPy below, triangle's constrained Delaunay triangulation
-        # returns triangles inside the hole as well. (I guess in case you want
-        # to render holes but in a different color, for example.) In our case,
-        # we want to get rid of them, so we cull them with some NumPy
-        # calculations
-        triangles = _cull_triangles_not_in_poly(
-            vertices, raw_triangles, polygon_vertices
-        )
-    else:
-        try:
-            vertices, triangles = PolygonData(
-                vertices=raw_vertices
-            ).triangulate()
-        except Exception as e:  # pragma: no cover
-            path, text_path = _save_failed_triangulation(raw_vertices)
-            raise RuntimeError(
-                f'Triangulation failed. Data saved to {path} and {text_path}'
-            ) from e
-
+    res = triangulate({'vertices': raw_vertices, 'segments': edges}, opts='p')
+    vertices = res['vertices']
+    raw_triangles = res['triangles']
+    # unlike VisPy below, triangle's constrained Delaunay triangulation
+    # returns triangles inside the hole as well. (I guess in case you want
+    # to render holes but in a different color, for example.) In our case,
+    # we want to get rid of them, so we cull them with some NumPy
+    # calculations
```

Comment:
```suggestion
    # triangle's constrained Delaunay triangulation
    # returns triangles inside the hole. (perhaps in case you want
    # to render holes but in a different color, for example.) In our case,
    # we want to get rid of them, so we cull them with some NumPy
    # calculations
```